ComputeAdmin/AzureStack.ComputeAdmin.psm1

```diff
@@ -660,5 +662,185 @@ Function CreateGalleyItem{
         Remove-Item "$workdir\MarketplaceItem.zip"
         $azpkg = '{0}\{1}' -f $workdir, $galleryItemName
         return Get-Item -LiteralPath $azpkg
+}    
+
+Function Add-VMExtension{
+
+    [CmdletBinding(DefaultParameterSetName='VMExtensionFromLocal')]
+    Param(
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtesionFromAzure')]
+        [ValidatePattern(“[a-zA-Z0-9-]{3,}”)]
+        [String] $publisher,
+
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtesionFromAzure')]
+        [ValidatePattern(“\d+\.\d+\.\d+”)]
+        [String] $version,
+
+        [Parameter(ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(ParameterSetName='VMExtesionFromAzure')]
+        [String] $type,
+
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtensionFromLocal')]
+        [ValidateNotNullorEmpty()]
+        [String] $extensionLocalPath,
+
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtesionFromAzure')]
+        [ValidateNotNullorEmpty()]
+        [String] $extensionBlobURI,
+
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtesionFromAzure')]
+        [ValidateSet('Windows' ,'Linux')]
+        [String] $osType,
+
+        [Parameter(ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(ParameterSetName='VMExtesionFromAzure')]
+        [String] $vmScaleSetEnabled = "false",
+
+        [Parameter(ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(ParameterSetName='VMExtesionFromAzure')]
+        [String] $supportMultipleExtensions = "false",
+    
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtesionFromAzure')]
+        [ValidateNotNullorEmpty()]
+        [String] $tenantID,
+
+        [Parameter(ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(ParameterSetName='VMExtesionFromAzure')]
+        [String] $location = 'local',
```

Comment:
It can be any string in the future. For now Azure stack supports only 1 region 'local' though.